pkg/sendgrid/sendgrid.go

```diff
@@ -174,6 +174,45 @@ func (c *Client) Delete(id string) error {
 	return nil
 }
 
+//Refresh deletes the API key associated with a subuser and generates a new key
+func (c *Client) Refresh(id string) (string, error) {
+	c.logger.Debugf("checking if sub user %s exists", id)
+	subuser, err := c.sendgridClient.GetSubUserByUsername(id)
+	if err != nil {
+		if IsNotExistError(err) {
+			return "", &smtpdetails.NotExistError{Message: err.Error()}
+		}
+		return "", errors.Wrapf(err, "check to see if sub user exists failed")
+	}
+	//required check? Debug message and continue?
+	if subuser.Username != id {
+		return "", errors.New(fmt.Sprintf("found user does not have expected username, expected=%s found=%s", id, subuser.Username))
+	}
+	c.logger.Debugf("sub user %s exists, finding user keys to check for key to delete", subuser.Username)
+	apiKeys, err := c.sendgridClient.GetAPIKeysForSubUser(subuser.Username)
+	if err != nil {
+		c.logger.Debugf(err.Error())
+	}
+	if len(apiKeys) > 0 {
+		for _, k := range apiKeys {
+			if k.Name == subuser.Username {
+				if err := c.sendgridClient.DeleteAPIKey(k.ID, k.Name); err != nil {
+					c.logger.Debugf(err.Error())
+					break
+				}
+				c.logger.Debugf("api key %s found and deleted", k.Name)
+				break
+			}
+		}
+	}
```

Comment:
Could something like:

```
// try to find the existing key
var foundKey *APIKey
for _, k := range apiKeys {
    if k.Name == subuser.Username {
        foundKey = k
        break
    }
}
// if a key exists, delete it
if foundKey != nil {
    if err = c.sendgridClient.DeleteAPIKey(k.ID, k.Name); err != nil {
        return "", errors.Wrapf(err, "failed to delete found api key, id=%s name=%s", k.ID, k.Name)
    }
}
// key is either removed or never existed, create it
apiKey, err := c.sendgridClient.CreateAPIKeyForSubUser(subuser.Username, DefaultAPIKeyScopes)
if err != nil {
    ...
}
return apiKey.Key, nil
```
provide a flatter structure here? This likely won't compile, I haven't tested it